gemet/thesaurus/views.py

```diff
@@ -223,17 +224,53 @@ def get_context_data(self, **kwargs):
             "namespace": Term.NAMESPACE,
             "status_values": self.status_values,
         })
+        if 'paginator' in kwargs:
+            context.update({
+                "paginator": kwargs['paginator']
+            })
         return context
 
-    def form_valid(self, form):
+    def form_valid(self, form, **kwargs):
         self.query = form.cleaned_data['query']
         self.concepts = search_queryset(
             self.query,
             self.language,
             status_values=self.status_values,
         )
+        page = self.request.GET.get('page', 1)
+        paginator = Paginator(self.concepts, 25)
+        self.concepts = paginator.page(page)
 
-        return self.render_to_response(self.get_context_data(form=form))
+        context = self.get_context_data(form=form, paginator=paginator)
+        page_number = self.concepts.number
+        total_pages = len(self.concepts.paginator.page_range)
+        distance_number = 25
```

Comment:
Why `25` for `distance`? The other views use `9`. Define a constant and use it in both places